pkg/sync/v6/client.go

```diff
@@ -254,3 +259,24 @@ func printDiff(log *logrus.Entry, existing, o *unstructured.Unstructured) bool {
 	}
 	return diffShown
 }
+
+// reconcileProtected check if object is marked with azure reconcile protection annotation
+// this will remove certain objects from being updated post cluster creation
+// access to modify these objects is limited by RBAC
+func (s *sync) reconcileProtected(existing, o *unstructured.Unstructured) bool {
+	if strings.ToLower(o.GetAnnotations()[reconcileProtectKeyAnnotationKey]) == "true" {
+		s.log.Infof("Sync reconcile protection detected for [%s/%s], ignore update", o.GetKind(), o.GetName())
+		return true
+	}
+	if strings.ToLower(existing.GetAnnotations()[reconcileProtectKeyAnnotationKey]) == "true" {
+		s.log.Infof("User reconcile protection detected for [%s/%s], ignore update", o.GetKind(), o.GetName())
+		// check sync pod managed/unmanaged dependencies
+		if (existing.GroupVersionKind().GroupKind() == schema.GroupKind{Kind: "Namespace"} &&
+			existing.GetName() == "openshift") {
+			s.reconcileConfig.managedSharedResource = false
```

Comment:
a) the lifecycle of s.reconcileConfig.managedSharedResource is going to confuse someone in the future.  Can you just explicitly get the namespace and inspect it in writeDB() and then decide whether to call applyResources(sharedResourceFilter) as a result?  That saves the pseudo-global spaghetti.
b) I think I'd rather see 2 annotations - one which prevents reconciliation of a given object, and a second one which is applied to the openshift project which prevents reconciliation of all objects within.  I don't think it makes sense to condense these into one, but am open to being persuaded.

Reply:
a) It makes logic little bit complex too, as now before applying `ImageStream` or `Template` I have to inspect each object to check which namespace it is in. So It can be done, but it will be case where we have to inspect all objects as as much as `sync` care its just `unstructured`. Im ok to change this. It looked cleaner this way in my head :)
b) I don't want to create 2 annotations as this one just does one thing. And in theory, it is the same thing. :man_shrugging: